### PR TITLE
[TFLite] Ensure that all int16x8 operators check that the zero-point is null

### PR DESCRIPTION
**Pull request by Tessil**
Hi,

The int16x8 operators should double-check that their int16 inputs or outputs don't use any zero point. Most int16x8 operators already check that but it wasn't the case for a few remaining kernels. This PR fixes that and adapt the tests accordingly to avoid the creation of int16 tensors with a non-null zero-point.

The int16 output type check in `PreluPrepare` is also removed as `PreluEval` doesn't support int16x8 quantization yet. I'm not sure why this check was added in commit c01a6d66514886d2d75f9f3d0d46301827d2b7e9.

Thanks,
Thibaut

### CODE REVIEW

**Comment by thaink** [comment]
Thanks for your contribution.
What is the bug associated with this PR?

**Comment by Tessil** [comment]
Hi,

There isn't any bug associate to it per se as an int16 tensor with a non-null zero-point should not be generated by the quantizer for now. It's mainly to keep coherence between the int16x8 versions of the operators we contributed and to explicitly state that they only need to support int16 tensors with a null zero-point.

**Comment by thaink** [comment]
@jianlijianli Could you help to review this PR?

File: tensorflow/lite/kernels/activations.cc [hunk 1]
```diff
@@ -678,8 +685,7 @@ TfLiteStatus PreluPrepare(TfLiteContext* context, TfLiteNode* node) {
 
   output->type = input->type;
 
-  if (output->type == kTfLiteUInt8 || output->type == kTfLiteInt8 ||
-      output->type == kTfLiteInt16) {
+  if (output->type == kTfLiteUInt8 || output->type == kTfLiteInt8) {
```
Comment (jianlijianli) [inline tensorflow/lite/kernels/activations.cc:688]: Could you please keep this Int16 support? I think this is added to support some use case.

Comment (Tessil) [reply tensorflow/lite/kernels/activations.cc:688]: It seems that `PreluEval` only support outputs of type `float`, `uint8_t` and `int8_t`. Do you know in which cases we could have a Prelu with an int16 output?

Comment (jianlijianli) [reply tensorflow/lite/kernels/activations.cc:688]: Ah, you are right, the eval function does not handle int16. Thanks for the change.

**Review by jianlijianli — APPROVED** [review]

**Pull request merged**